### PR TITLE
OCPQE-20187 OCPQE-20188 Support slack alert and skip optional steps when job is success

### PR DESCRIPTION
**Pull request by rioliu-rh**
- Enable slack notification for MCO jobs
- Skip optional steps e.g. data gathering when job is success

### CODE REVIEW

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-aws-ipi-f9-longduration-mco-critical

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-aws-ipi-f9-longduration-mco-critical

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-aws-ipi-f9-longduration-mco-critical

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-aws-ipi-f9-longduration-mco-critical

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__amd64-nightly.yaml [hunk 1]
```diff
@@ -265,6 +266,7 @@ tests:
 - as: aws-ipi-f9-longduration-mco-all
   cron: 3 17 6,18,27 * *
   steps:
+    allow_skip_on_success: true
```
Comment (kuiwang02) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__amd64-nightly.yaml:269]: @rioliu-rh if it is true, post steps are not executed when pre and test step are successful.

but we need to execute post to release resource, so please do not set it as true.

Comment (kuiwang02) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__amd64-nightly.yaml:269]: maybe it is ok if there is no `optional_on_success: true` in step of post for removing resource.

Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__amd64-nightly.yaml:269]: Example working [log](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/logs/periodic-ci-openshift-openshift-tests-private-release-4.15-automated-release-aws-ipi-fips-f1/1770964040256851968),
```
......
INFO[2024-03-22T02:22:58Z] Step aws-ipi-fips-f1-openshift-e2e-test-qe-report succeeded after 8s. 
INFO[2024-03-22T02:22:58Z] Step phase test succeeded after 1h39m30s.    
INFO[2024-03-22T02:22:58Z] Running multi-stage phase post               
INFO[2024-03-22T02:22:58Z] Skipping optional step aws-ipi-fips-f1-gather-network 
INFO[2024-03-22T02:22:58Z] Skipping optional step aws-ipi-fips-f1-gather-core-dump 
INFO[2024-03-22T02:22:58Z] Skipping optional step aws-ipi-fips-f1-gather-aws-console 
INFO[2024-03-22T02:22:58Z] Skipping optional step aws-ipi-fips-f1-gather-must-gather 
INFO[2024-03-22T02:22:58Z] Skipping optional step aws-ipi-fips-f1-gather-extra 
INFO[2024-03-22T02:22:58Z] Skipping optional step aws-ipi-fips-f1-gather-audit-logs 
INFO[2024-03-22T02:22:58Z] Running step aws-ipi-fips-f1-ipi-deprovision-deprovision. 
INFO[2024-03-22T02:30:32Z] Step aws-ipi-fips-f1-ipi-deprovision-deprovision succeeded after 7m33s. 
INFO[2024-03-22T02:30:32Z] Step phase post succeeded after 7m33s. 
```

Comment (kuiwang02) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__amd64-nightly.yaml:269]: Thanks for info.

Comment (rioliu-rh) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__amd64-nightly.yaml:269]: @kuiwang02 which post step will be skipped, as I know only the optional (has `optional_on_success: true`) step will be skipped when this attribute is true.

Comment (rioliu-rh) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__amd64-nightly.yaml:269]: ref: https://docs.ci.openshift.org/docs/architecture/step-registry/#skipping-post-steps-on-success

**Comment by rioliu-rh** [comment]
@liangxia the rehearse job is failed, but I cannot receive the slack alert, is there any config I missed?

**Comment by liangxia** [comment]
> @liangxia the rehearse job is failed, but I cannot receive the slack alert, is there any config I missed?

Jobs triggered in PR does not get notification.

**Comment by liangxia** [comment]
Let's merge this PR.
@rioliu-rh You can give it another try after the PR merge.

**Comment by rioliu-rh** [comment]
@sergiordlr @liangxia @kuiwang02 
I got a successful job, gathering steps are skipped when job is success
https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/logs/periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-aws-ipi-f9-longduration-mco-critical/1771056548978626560
```console
INFO[2024-03-22T10:41:48Z] Running multi-stage phase post               
INFO[2024-03-22T10:41:48Z] Skipping optional step aws-ipi-f9-longduration-mco-critical-gather-network 
INFO[2024-03-22T10:41:48Z] Skipping optional step aws-ipi-f9-longduration-mco-critical-gather-core-dump 
INFO[2024-03-22T10:41:48Z] Skipping optional step aws-ipi-f9-longduration-mco-critical-gather-aws-console 
INFO[2024-03-22T10:41:48Z] Skipping optional step aws-ipi-f9-longduration-mco-critical-gather-must-gather 
INFO[2024-03-22T10:41:48Z] Skipping optional step aws-ipi-f9-longduration-mco-critical-gather-extra 
INFO[2024-03-22T10:41:48Z] Skipping optional step aws-ipi-f9-longduration-mco-critical-gather-audit-logs 
INFO[2024-03-22T10:41:48Z] Running step aws-ipi-f9-longduration-mco-critical-ipi-deprovision-deprovision. 
INFO[2024-03-22T10:49:33Z] Step aws-ipi-f9-longduration-mco-critical-ipi-deprovision-deprovision succeeded after 7m44s. 
```
slack alert is received.
thanks

**Pull request merged**